src/main/java/seedu/address/logic/commands/CheckCraftCommand.java

```diff
@@ -0,0 +1,144 @@
+
+/**
+ * Checks if there are recipes which may be used to craft the desired quantity of the item.
+ */
+public class CheckCraftCommand extends Command {
+    public static final String COMMAND_WORD = "check";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Checks if the item may be crafted\n"
+            + "Parameters: "
+            + PREFIX_ITEM_NAME + "ITEM NAME "
+            + PREFIX_ITEM_QUANTITY + "ITEM QUANTITY \n"
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_ITEM_NAME + "Bob's blueberry "
+            + PREFIX_ITEM_QUANTITY + "20";
+
+    public static final String MESSAGE_SUCCESS_CRAFTABLE = "%2$s %1$s may be crafted with any of the recipes below:";
+    public static final String MESSAGE_SUCCESS_UNCRAFTABLE = "%2$s %1$s is not possible to craft with a single recipe";
+
+    public static final String MESSAGE_INVALID_QUANTITY = "Please enter a valid quantity";
+    public static final String MESSAGE_ITEM_NOT_FOUND = "Item to craft is not found in the item list.";
+    public static final String MESSAGE_RECIPE_NOT_FOUND = "No recipes to craft this item found in the recipe list";
+
+    private final String itemName;
+    private final int quantity;
+
+    /**
+     * Creates a CheckCraftCommand to check if the {@code quantity} of the item with {@code itemName} may be crafted.
+     */
+    public CheckCraftCommand(String itemName, Quantity quantity) {
+        this.itemName = itemName;
+        this.quantity = quantity.getNumber();
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        if (quantity == 0) {
+            throw new CommandException(MESSAGE_INVALID_QUANTITY);
+        }
+
+        // fetch the item
+        List<Item> tempItemList = new ArrayList<>(model.getFilteredItemList());
+        // filter to only get matching items
+        tempItemList.removeIf(x -> !x.getName().equals(itemName));
+        if (tempItemList.isEmpty()) {
+            throw new CommandException(MESSAGE_ITEM_NOT_FOUND);
+        }
+        Item item = tempItemList.stream()
+                .findFirst() // Get the first (and only) item matching or else throw Error
+                .orElseThrow(() -> new CommandException(MESSAGE_ITEM_NOT_FOUND));
+
+        ArrayList<Recipe> recipeList = new ArrayList<>(model.getFilteredRecipeList());
+        // filter to only get matching recipes
+        recipeList.removeIf(x -> !x.getProductName().equals(itemName));
+        if (recipeList.isEmpty()) {
+            throw new CommandException(MESSAGE_RECIPE_NOT_FOUND);
+        }
+
+        ArrayList<Item> itemList = new ArrayList<>(model.getFilteredItemList());
+
+        return new CommandResult(findRecipes(quantity, itemList, recipeList));
+    }
+
+    /**
+     * Extracts all usable recipes and formats them for display.
+     */
+    private String findRecipes(int quantity, ArrayList<Item> itemList, ArrayList<Recipe> recipeList) {
```

Comment:
Very Slight Nitpick but generally better to use interface in the argument rather than an ArrayList<T>